src/plugins/info/plugin.py

```diff
@@ -103,6 +110,57 @@ def on_config(self, config):
             log.error("Please remove 'hooks' setting.")
             self._help_on_customizations_and_exit()
 
+        # Assure that possible relative paths, which will be validated
+        # or used to generate other paths are absolute.
+        config.config_file_path = _convert_to_abs(config.config_file_path)
+        config_file_parent = os.path.dirname(config.config_file_path)
+
+        # The theme.custom_dir property cannot be set, therefore a helper
+        # variable is used.
+        custom_dir = config.theme.custom_dir
+        if custom_dir:
+            custom_dir = _convert_to_abs(
+                custom_dir,
+                abs_prefix = config_file_parent
+            )
+
+        # Support projects plugin
+        projects_plugin = config.plugins.get("material/projects")
+        if projects_plugin:
+            abs_projects_dir = _convert_to_abs(
+                projects_plugin.config.projects_dir,
+                abs_prefix = config_file_parent
+            )
+        else:
+            abs_projects_dir = ""
+
+        # Load the current MkDocs config(s) to get access to INHERIT
```

Comment:
Could you add some more info in the comments on the different design decisions? We discussed the implementation quite a lot, and our decisions and the limitations those decisions try to workaround should be commented here. You can check other plugins to see how I'm doing this, because when somebody wants to refactor the code, the implications should be clear.

Example:

https://github.com/squidfunk/mkdocs-material/blob/15f15068d958506734d3544108a1e6971d7e511b/src/plugins/blog/plugin.py#L171-L184

Reply:
I think I elaborated a bit better, but I'm not sure if it's resolved, so let me know if not 👌

Reply:
Looks good ☺️